transports/tcp/src/provider.rs

```diff
@@ -46,6 +48,14 @@ pub trait Provider: Clone + Send + 'static {
     type Stream: AsyncRead + AsyncWrite + Send + Unpin + fmt::Debug;
     /// The type of TCP listeners obtained from [`Provider::new_listener`].
     type Listener: Send + Unpin;
+    /// The type of IfWatcher obtained from [`Provider::new_if_watcher`].
+    type IfWatcher: Stream<Item = io::Result<IfEvent>> + Send + Unpin;
```

Comment:
```suggestion
    type IfWatcher: Stream<Item = io::Result<IfEvent>> + IntoIterator<Item = IpNet> + Send + Unpin;
```

Is this legal Rust? My brain compiler is unsure but it could avoid the need for the `addrs` function I think.

Reply:
It is legal syntax, but neither IfWatcher implementation implements this trait, plus `into_iter` consumes `self`, so we wouldn’t be able to use this here.

Reply:
Ah damn!

The naming of `iter` suggested that we actually implement the `Iterator` trait but I guess the consuming nature of `IntoIterator` makes this unworkable anyway.

Thanks for checking!